ecr-login/api/client.go

```diff
@@ -98,6 +99,22 @@ func (self *defaultClient) GetCredentials(registry, image string) (*Auth, error)
 	return nil, fmt.Errorf("No AuthorizationToken found for %s", registry)
 }
 
+func (self *defaultClient) ListCredentials() ([]*Auth, error) {
+	log.Debug("Listing Credentials from cache")
+
+	auths := []*Auth{}
+	for _, authEntry := range self.credentialCache.List() {
```

Comment:
Since you're just walking the cache, are you going to return expired credentials?  Should we refresh credentials for those that are expired so that valid entries can be returned?